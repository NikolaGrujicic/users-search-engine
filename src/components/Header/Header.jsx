import React from 'react';
import { Link } from 'react-router-dom';
import SearchBar from '../SearchBar/SearchBar';

const Header = () => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top">
    <a className="navbar-brand" href="/">Users Search Engine</a>
    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <span className="navbar-toggler-icon" />
    </button>

    <div className="collapse navbar-collapse" id="navbarSupportedContent">
      <ul className="navbar-nav mr-auto">
        <li className="nav-item active">
          <Link className="nav-link" to="/users">
            Users
            <span className="sr-only">(current)</span>
          </Link>
        </li>
        <li className="nav-item active">
          <Link className="nav-link" to="/settings">
            Settings
            <span className="sr-only">(current)</span>
          </Link>
        </li>
      </ul>
      <SearchBar />
    </div>
  </nav>
);

export default Header;
