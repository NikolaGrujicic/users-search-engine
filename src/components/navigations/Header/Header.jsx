import React from 'react';
import { AppBar, Toolbar, Typography } from '@material-ui/core';
import useStyles from '../../../styles/muiStyles';

const TopNav = () => {
  const classes = useStyles();

  return (
    <AppBar position="fixed" className={classes.appBar}>
      <Toolbar>
        <Typography variant="h6" noWrap>
          Clipped drawer
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

export default TopNav;
