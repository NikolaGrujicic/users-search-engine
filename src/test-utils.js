import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { applyMiddleware, compose, createStore } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import rootReducer from './redux/rootReducer';

const middlewares = [thunk];

const enhancers = [applyMiddleware(...middlewares)];

const composeEnhancers = process.env.NODE_ENV !== 'production'
  && typeof window === 'object'
  && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
  ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
  : compose;

function render(
  ui,
  {
    initialState,
    store = createStore(rootReducer, initialState, composeEnhancers(...enhancers)),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const renderWithRouter = (ui, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return rtlRender(ui, { wrapper: Router });
};

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { render, renderWithRouter };
