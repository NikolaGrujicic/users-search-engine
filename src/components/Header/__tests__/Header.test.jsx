import React from 'react';
import { screen, renderWithRouter } from '../../../test-utils';
import Header from '../Header';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

test('renders Header', () => {
  renderWithRouter(<Header />);

  expect(screen.queryByText(/Address book/i)).toBeInTheDocument();
  expect(screen.queryByText(/Users/i)).toBeInTheDocument();
  expect(screen.queryByText(/Settings/i)).toBeInTheDocument();
});
