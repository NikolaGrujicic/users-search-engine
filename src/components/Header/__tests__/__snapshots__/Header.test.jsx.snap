// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark fixed-top"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Users Search Engine
  </a>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/"
        >
          Home
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          to="/users"
        >
          Users
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          to="/settings"
        >
          Settings
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
    <SearchBar />
  </div>
</nav>
`;
